src/http.rs

```diff
@@ -61,3 +72,87 @@ fn add_auth_headers<'a>(headers: &'a mut HeaderMap, user: &GlobalUser) {
         }
     }
 }
+
+////---------------------------NEW API CLIENT CODE---------------------------////
+pub fn api_client(
+    user: &GlobalUser,
+    config: HttpApiClientConfig,
+) -> Result<HttpApiClient, failure::Error> {
+    HttpApiClient::new(
+        Credentials::from(user.to_owned()),
+        config,
+        Environment::Production,
+    )
+}
+
+// This enum allows callers of format_error() to specify a help_*() function for adding additional
+// detail to error messages, if desired.
+pub enum ErrorCodeDetail {
+    None,
+    WorkersKV,
+}
+
+// Format errors from the cloudflare-rs cli for printing.
+pub fn format_error(e: ApiFailure, err_type: ErrorCodeDetail) -> String {
+    match e {
+        ApiFailure::Error(status, api_errors) => {
+            print_status_code_context(status);
+            let mut complete_err = "".to_string();
+            for error in api_errors.errors {
+                let error_msg =
+                    format!("{} Error {}: {}\n", emoji::WARN, error.code, error.message);
+
+                let suggestion = match err_type {
+                    ErrorCodeDetail::WorkersKV => kv_help(error.code),
+                    _ => "",
+                };
+                if !suggestion.is_empty() {
+                    let help_msg = format!("{} {}\n", emoji::SLEUTH, suggestion);
+                    complete_err.push_str(&format!("{}{}", error_msg, help_msg));
+                } else {
+                    complete_err.push_str(&error_msg)
+                }
+            }
+            complete_err
+        }
+        ApiFailure::Invalid(reqwest_err) => format!("{} Error: {}", emoji::WARN, reqwest_err),
+    }
+}
+
+// For handling cases where the API gateway returns errors via HTTP status codes
+// (no API-specific, more granular error code is given).
+fn print_status_code_context(status_code: StatusCode) {
+    match status_code {
+        // Folks should never hit PAYLOAD_TOO_LARGE, given that Wrangler ensures that bulk file uploads
+        // are max ~50 MB in size. This case is handled anyways out of an abundance of caution.
+        StatusCode::PAYLOAD_TOO_LARGE =>
+            message::warn("Returned status code 413, Payload Too Large. Please make sure your upload is less than 100MB in size"),
+        StatusCode::GATEWAY_TIMEOUT =>
+            message::warn("Returned status code 504, Gateway Timeout. Please try again in a few seconds"),
+        _ => (),
+    }
+}
+
+// kv_help() provides more detailed explanations of Workers KV API error codes.
+// See https://api.cloudflare.com/#workers-kv-namespace-errors for details.
+fn kv_help(error_code: u16) -> &'static str {
```

Comment:
I opened this issue a while back that asks that we have all of our API codes handled once we move to cloudflare-rs - if we take the passing function approach, could @gabbifish add a comment on this? https://github.com/cloudflare/wrangler/issues/740

Reply:
yeah good call; for posterity's sake i'll clarify that the reason they need to be separate is, unfortunately, the different APIs behind the gateway have overlapping error codes, so we can't reliably match on them everywhere; that is, both Workers and Workers KV APIs have an error code 10000 that mean drastically different things.

Reply:
> I figured it would be sort of nice to have distinct error codes explanations for all types of endpoints in one place (readability)

i think at the top level, or in its own directory, per api (i.e. workers, workers kv), is a decent level of consolidation.